pandas/_libs/reduction.pyx

```diff
@@ -354,7 +348,7 @@ def apply_frame_axis0(object frame, object f, object names,
             slider.move(starts[i], ends[i])
 
             item_cache.clear()  # ugh
-            chunk = slider.dummy
+            chunk = slider.frame[starts[i]:ends[i]]
```

Comment:
Can't you just use the frame here, and delete the whole `BlockSlider` class? E.g. https://github.com/ivirshup/pandas/commit/2c17e72a9deffd8e0d8999b88a2e31c0a58c746c ? I don't get any change in test failures from this.

Reply:
I tried this but still saw test failures - did you do that in combination with the changes to `.data` assignment that are already here?

Reply:
If you have an alternate approach it's also worth pushing up as a separate PR. Might be easier to review that way

Reply:
> I tried this but still saw test failures

The commit I referenced is just one commit on top of bce72ac646fbbe from this PR. I get the same test failures from that as on bce72ac646fbbe. I just wanted to note if you go with this approach, you can delete a bunch of other code.

I don't think I know enough about cython or pandas internals to be too much help here. I've just played around with this code, but am still working on actually understanding it.